pkg/networkservice/common/refresh/client.go

```diff
@@ -74,17 +75,25 @@ func (t *refreshClient) Request(ctx context.Context, request *networkservice.Net
 	store(ctx, metadata.IsClient(t), cancel)
 
 	eventFactory := begin.FromContext(ctx)
-	timeClock := clock.FromContext(ctx)
+	clockTime := clock.FromContext(ctx)
 	// Create the afterCh *outside* the go routine.  This must be done to avoid picking up a later 'now'
 	// from mockClock in testing
-	afterCh := timeClock.After(refreshAfter)
-	go func(cancelCtx context.Context, afterCh <-chan time.Time) {
-		select {
-		case <-cancelCtx.Done():
-		case <-afterCh:
-			eventFactory.Request(begin.CancelContext(cancelCtx))
+	afterTicker := clockTime.Ticker(refreshAfter)
+	go func() {
+		defer afterTicker.Stop()
+		for {
+			select {
+			case <-cancelCtx.Done():
+				return
+			case <-afterTicker.C():
+				if err := <-eventFactory.Request(begin.CancelContext(cancelCtx)); err != nil {
+					logger.Warnf("refresh failed: %s", err.Error())
+					continue
+				}
+				return
+			}
```

Comment:
Why retry instead of letting the next refresh catch it?

Reply:
So... perhaps I'm simply confused... but this looks to me like it will call the refresh *once*, that will cancel the context (see line 72 above) and then it will return based on line 86 ... meaning... I don't see what this does for us at all.  I can see some of the other fixes in the PR... but this just introduces complexity I don't think will ever be used...

Reply:
Ah wait... I see.. the context isn't canceled if our call to the downstream gets an error.

Reply:
OK... this is clever... it will tick at the roughly 1/3 marks even if we get an error... so not overwhelming... but keeps trying.